spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/audit/domain/AuditRecord.java

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Represents an audit entry. Used to record audit trail information.
+ *
+ * @author Gunnar Hillert
+ */
+@Entity
+@Table(name = "AUDIT_RECORDS")
+@EntityListeners(AuditingEntityListener.class)
+public class AuditRecord {
+
+	private static final String UNKNOW_SERVER_HOST = "unknown";
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	private Long id;
+
+	@Column(name = "created_by")
+	@CreatedBy
+	private String createdBy;
+
+	@Column(name = "server_host")
+	private String serverHost;
+
+	@Column(name = "correlation_id")
+	private String correlationId;
+
+	@Lob
+	@Type(type="org.hibernate.type.TextType")
+	@Column(name = "audit_data")
+	private String auditData;
+
+	@CreatedDate
+	private Instant createdOn;
+
+	@NotNull
+	@Type(type = "org.springframework.cloud.dataflow.server.hibernate.GenericEnumUserType", parameters = {
+			@Parameter(name = "enumClass", value = "org.springframework.cloud.dataflow.server.audit.domain.AuditActionType"),
+			@Parameter(name = "identifierMethod", value = "getId"),
+			@Parameter(name = "valueOfMethod", value = "fromId") })
+	private AuditActionType auditAction;
+
+	@NotNull
+	@Type(type = "org.springframework.cloud.dataflow.server.hibernate.GenericEnumUserType", parameters = {
+			@Parameter(name = "enumClass", value = "org.springframework.cloud.dataflow.server.audit.domain.AuditOperationType"),
+			@Parameter(name = "identifierMethod", value = "getId"),
+			@Parameter(name = "valueOfMethod", value = "fromId") })
+	private AuditOperationType auditOperation;
+
+	public Long getId() {
+		return id;
+	}
+
+	public void setId(Long id) {
+		this.id = id;
+	}
+
+	public String getCreatedBy() {
+		return createdBy;
+	}
+
+	public void setCreatedBy(String createdBy) {
+		this.createdBy = createdBy;
+	}
+
+	public Instant getCreatedDateTime() {
+		return createdOn;
+	}
+
+	public void setCreatedOn(Instant createdOn) {
+		this.createdOn = createdOn;
+	}
+
+	public AuditActionType getAuditAction() {
+		return auditAction;
+	}
+
+	public void setAuditAction(AuditActionType auditAction) {
+		this.auditAction = auditAction;
+	}
+
+	public AuditOperationType getAuditOperation() {
+		return auditOperation;
+	}
+
+	public void setAuditOperation(AuditOperationType auditOperation) {
+		this.auditOperation = auditOperation;
+	}
+
+	public String getServerHost() {
+		return serverHost;
+	}
+
+	public String getCorrelationId() {
+		return correlationId;
+	}
+
+	public void setCorrelationId(String correlationId) {
+		this.correlationId = correlationId;
+	}
+
+	public String getAuditData() {
+		return auditData;
+	}
+
+	public void setAuditData(String data) {
+		this.auditData = data;
+	}
+
+	public Instant getCreatedOn() {
+		return createdOn;
+	}
+
+	/**
+	 * Automatically populate the server host for auditing purposes.
+	 */
+	@PrePersist
+	@PreUpdate
+	@PreRemove
+	public void populateServerHost() {
+		try {
+			this.serverHost = InetAddress.getLocalHost().getHostName();
```

Comment:
We should discuss the server host field.  I had originally thought of it to be recording the dataflow server name, something that would be easily connected to the users/admin of the system.  Might still be useful to have the hostname, but in the cloud environment, that is often not so useful.  Other fields to consider would be something about the cloud provider, in the case of cloud foundry, the org/space or namespace in k8s.  For skipper deployments we would record the `account` that was being targeted.

Reply:
If we want a host, using the Boot property `spring.cloud.client.hostname` is a bit more robust than the code here.  See https://github.com/spring-cloud/spring-cloud-commons/blob/master/spring-cloud-commons/src/main/java/org/springframework/cloud/client/HostInfoEnvironmentPostProcessor.java

Reply:
that code turns out to not be so useful in the context of a domain class.